src/shogun/multiclass/KNN.cpp

```diff
@@ -104,7 +105,7 @@ bool KNN::train_machine(std::shared_ptr<Features> data)
 	return true;
 }
 
-SGMatrix<index_t> KNN::nearest_neighbors()
+SGMatrix<index_t> KNN::nearest_neighbors(KNN_SOLVER knn_solver/*=KNN_BRUTE*/)
```

Comment:
not sure I understand ...

Reply:
I converted the `nearest_neighbour()` unparameterized function to one with a single parameter of type `KNN_SOLVER`. The `/*=KNN_BRUTE*/` is just to let the reader know that the default value is `KNN_BRUTE` as stated in the `KNN.h` file.